### ISSUE TITLE
Code that *should* fail MYFIRSTIO?

### ISSUE THREAD

**Issue by mdouglasbrett** [nodeschool/discussions#875]
Hi guys!

I'm a big fan of your program. Just wanted to raise something. The following code, passes the verifier:
    
    var fs = require('fs');
    var buff = fs.readFileSync( '/Users/michaelbrett/learnyounode/myfirstio.js' );
    var str = buff.toString();
    var arr = str.split('\n');

    console.log(arr.length);

But after seeing the actual solution, I realised I had missed the point of using process.argv. In this case, I hadn't truly fulfilled the aim of the exercise - Is it too picky to make the exercise demand the use of process.argv to be successful?

Just a thought.

Keep up the good work!

Mike

**Comment by martinheidegger** [comment]
@mdouglasbrett Good point. Certainly worth an improvement to [learnyounode](https://github.com/rvagg/learnyounode)! Can you copy the issue to there or even better work on a PR? I am sure it will be taken gladly. (There are often people available in https://gitter.im/rvagg/learnyounode to talk about things)

**Comment by polizz** [comment]
I had the same code as Mike, but it would not pass. I scratched my head for about 20 minutes trying to figure out why there was such a huge line count discrepancy, and finally printed out the argv contents from within the test program to see that there was a dynamic file parameter being sent in. I think I spent more time troubleshooting what the exercise was doing rather than the topic of file I/O.

Overall, I find this way of learning enjoyable though, so keep up the good fight.

Andrew
